### PR TITLE
Mrc 448 allfilters

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Adding the age, survey and and region filters. Some added complexity because the region filters are potentially nested, so some recursion required.

### CODE REVIEW

**Comment by EmmaLRussell** [comment]
The code coverage is moaning because of the fake data that's being added if none comes back from the API. Not writing a test for that as it will be gone soon!

File: src/app/static/src/app/components/Filters.vue [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+                    <label>Survey</label>
+                    <treeselect id="survey-filters" :multiple="true"
+                                :options="surveyFilters.available"
+                                :value="surveyFilters.selected"
+                                :normalizer = "treeselectNormalizer"
+                                @input="updateSurveyFilter"></treeselect>
+                </div>
+                <div class="col-sm-3">
+                    <label>Region</label>
+                    <treeselect id="survey-filters" :multiple="true"
```
Comment (r-ash) [inline src/app/static/src/app/components/Filters.vue]: This ID should be `region-filters`?

File: src/app/static/src/scss/style.scss [hunk 2]
```diff
@@ -108,8 +113,16 @@ a {
 }
 
 .vue-treeselect__label-container, .vue-treeselect__label-container:hover {
-  .vue-treeselect__checkbox--checked {
-    border-color: $theme-red;
-    background: $theme-red;
+  .vue-treeselect__checkbox--checked, .vue-treeselect__checkbox--indeterminate {
+    border-color: $theme-red !important;
```
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply src/app/static/src/scss/style.scss]: Seemed to be necessary here to overcome the treeselect component's urge to make everything blue. We don't define the innards of the components so we don't have the option of putting additional classes on these sub-elements, so not sure there's much alternative.

File: src/app/static/src/app/components/Filters.vue (hunk 1 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/components/Filters.vue]: Yup!

**Comment by EmmaLRussell** [comment]
>Should we hide the survey filter on all tabs except for the survey page? (at the moment these filters don't belong to a tab do they?)

We've got a separate ticket to make a visual indication of what filters are relevant for the currently displayed data. Maybe hiding the survey component, or maybe just greying it out (might be more odd if it just vanished).

**Comment by EmmaLRussell** [comment]
> In the UI there isn't a way to switch between uploaded data - that will be a separate PR?

yep, tabs will be in a separate ticket

**Comment by EmmaLRussell** [comment]
>  Or we should keep these filters but have some visual cue they are unavailable (grey them out or something)

Yeah, I thought we'd decided to keep any filters that the user has selected, even if some of them aren't currently relevant. (Don't lose them when switch back.) Future ticket to grey or otherwise indicate that these tickets aren't currently available.)

**Comment by r-ash** [comment]
> > Or we should keep these filters but have some visual cue they are unavailable (grey them out or something)
> 
> Yeah, I thought we'd decided to keep any filters that the user has selected, even if some of them aren't currently relevant. (Don't lose them when switch back.) Future ticket to grey or otherwise indicate that these tickets aren't currently available.)

That sounds good.

**Comment by EmmaLRussell** [comment]
Made the Programme change, updated the duff ID, made the stepper centred, with a horizontal rule, and tweaked the height of the filter controls, so hopefully it looks a little tidier!

**Comment by r-ash** [comment]
Still reads "program" in the stepper (sorry)

**Comment by EmmaLRussell** [comment]
> Still reads "program" in the stepper (sorry)

:woman_facepalming:

**Pull request merged**